pkg/apis/kibana/v1/webhook.go

```diff
@@ -104,3 +107,12 @@ func checkSupportedVersion(k *Kibana) field.ErrorList {
 func checkNoDowngrade(prev, curr *Kibana) field.ErrorList {
 	return commonv1.CheckNoDowngrade(prev.Spec.Version, curr.Spec.Version)
 }
+
+func checkMonitoring(k *Kibana) field.ErrorList {
+	errs := validations.Validate(k, k.Spec.Version)
+	if monitoring.IsDefined(k) && !k.Spec.ElasticsearchRef.IsDefined() {
+		errs = append(errs, field.Invalid(field.NewPath("spec").Child("elasticsearchRef"), k.Spec.ElasticsearchRef,
+			"Monitoring does not support Kibana without associated Elasticsearch"))
```

Comment:
something a bit more explicit: `Kibana must be associated to an Elasticsearch cluster through elasticsearchRef in order to enable monitoring features`

Reply:
We should also call this function as part of the regular Kibana reconciliation?

Reply:
Yes and we already do it when we call `validate`:

https://github.com/elastic/cloud-on-k8s/blob/f52ca4700f56238c56186f0fc751ada563cf6dda/pkg/controller/kibana/controller.go#L187